app/gosns/confirm_subscription.go

```diff
@@ -0,0 +1,40 @@
+package gosns
+
+import (
+	"net/http"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func ConfirmSubscriptionV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+	requestBody := models.NewConfirmSubscriptionRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+	if !ok {
+		log.Error("Invalid Request - ConfirmSubscriptionV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+	topicArn := requestBody.TopicArn
+	confirmToken := requestBody.Token
+	var pendingConfirm pendingConfirm
+
+	if pending, ok := TOPIC_DATA[topicArn]; !ok {
+		return utils.CreateErrorResponseV1("SubscriptionNotFound", false)
+	} else {
+		pendingConfirm = *pending
+	}
+
+	if pendingConfirm.token != confirmToken {
+		return utils.CreateErrorResponseV1("SubscriptionNotFound", false)
+	}
+	respStruct := models.ConfirmSubscriptionResponse{
+		Xmlns:    models.BASE_XMLNS,
+		Result:   models.ConfirmSubscriptionResult{SubscriptionArn: pendingConfirm.subArn},
+		Metadata: app.ResponseMetadata{RequestId: uuid.NewString()},
+	}
+	return http.StatusOK, respStruct
```

Comment:
As far as we looked into current impl, current ConfirmSubscription do nothing actually since the subscriptions in goaws won't be "pending" status even if the subscribe endpoint will be an email address or a HTTP url.

Reply:
Yeah I think this endpoint is there as a functionality nicety.  I haven't looked at this in AWS too closely but it must double check on the HTTP subscriptions.  While it's not "doing" anything, you're right, it is just kind of checking giving you a check that we have a record of the subscription.  I guess that's worthy? 🤷

Reply:
Oh, sorry I didn't want to say to remove this API. 
I think the existence of this API on goaws is valuable in itself. Without it, ConfirmSubscription requests from user's applications would return 404 and it would not be able to fulfill its role as a stub.
Please get my above comment was for to be on the same page about current implementation 👍

Reply:
Yeah, I think we're saying the same thing in different ways.  I think it's fine that there's no real pending status, we'll just do what we do here.  The word "pending" in the code, I think is also still fine here, since it shadows what would probably be case in real life.  So all good, as far as I'm concerned!